"panic: interface conversion" when doing "zapi help create" or "zapi help find"

I was trying to verify #882 and hence working at `zq` commit `2bbd3c9`, which is the commit for linked PR #1289. I tried to look at the help text and got a panic.

```
$ zapi help create
panic: interface conversion: charm.Command is *cmd.Command, not *idx.IndexCmd

goroutine 1 [running]:
github.com/brimsec/zq/cmd/zapi/cmd/index.NewCreate(0x1b2f700, 0xc0003ec080, 0xc0003f6120, 0xc0003f6120, 0x6, 0xc00029fc01, 0x2)
	/Users/phil/work/zq/cmd/zapi/cmd/index/create.go:56 +0x27a
github.com/mccanne/charm.newInstance(0x1b2f700, 0xc0003ec080, 0x20e4300, 0x20e4300, 0x1, 0x7ffeefbffaaa)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:19 +0x74
github.com/mccanne/charm.(*HelpCommand).search(0xc0003f2004, 0xc0000321d0, 0x1, 0x1, 0x0, 0x100eac8, 0x0, 0xc00006c800, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/help.go:74 +0x1e7
github.com/mccanne/charm.(*HelpCommand).Run(0xc0003f2004, 0xc0000321d0, 0x1, 0x1, 0xc0000321d0, 0x1)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/help.go:87 +0x4d
github.com/mccanne/charm.(*instance).run(0xc0003f4020, 0xc0000321d0, 0x1, 0x1, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).Exec(0x20e4a80, 0x1b2f700, 0xc0003ec000, 0xc0000321d0, 0x1, 0x1, 0x2, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:67 +0x92
github.com/mccanne/charm.(*instance).run(0xc0003f4000, 0xc0000321c0, 0x2, 0x2, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:61 +0x41d
github.com/mccanne/charm.(*Spec).ExecRoot(0x20e41c0, 0xc0000321c0, 0x2, 0x2, 0xc00029ff78, 0x100763f, 0xc00009c058, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/phil/work/zq/cmd/zapi/main.go:20 +0x78
```

I see a similar panic when I do `zapi help find`.

--- a/cmd/zapi/cmd/index/create.go
+++ b/cmd/zapi/cmd/index/create.go
@@ -39,10 +39,6 @@ zapi index create -k id.orig_h -k count -o custom -z "count() by _path, id.orig_
 	New: NewCreate,
 }
 
-func init() {
-	cmd.CLI.Add(Create)
-}
-
 type CreateCmd struct {
 	*cmd.Command
 	root       string
@@ -53,7 +49,7 @@ type CreateCmd struct {
 }
 
 func NewCreate(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
-	c := &CreateCmd{Command: parent.(*IndexCmd).Command}
+	c := &CreateCmd{Command: parent.(*Command).Command}
 	f.Var(&c.keys, "k", "key fields (can be specified multiple times)")
 	f.StringVar(&c.inputFile, "i", "", "input file relative to each zar directory ('' means archive log file in the parent of the zar directory)")
 	f.StringVar(&c.outputFile, "o", "index.zng", "name of microindex output file (for custom indexes)")

--- a/cmd/zapi/cmd/index/find.go
+++ b/cmd/zapi/cmd/index/find.go
@@ -50,10 +50,6 @@ records of the base layer of the index file (-z)
 	New: NewFind,
 }
 
-func init() {
-	cmd.CLI.Add(Find)
-}
-
 type FindCmd struct {
 	*cmd.Command
 	indexFile     string
@@ -64,7 +60,7 @@ type FindCmd struct {
 }
 
 func NewFind(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
-	c := &FindCmd{Command: parent.(*IndexCmd).Command}
+	c := &FindCmd{Command: parent.(*Command).Command}
 	f.StringVar(&c.indexFile, "x", "", "name of microindex for custom index searches")
 	f.StringVar(&c.pathField, "l", archive.DefaultAddPathField, "zng field name for path name of log file")
 	f.BoolVar(&c.relativePaths, "relative", false, "display paths relative to root")

--- a/cmd/zapi/cmd/index/index.go
+++ b/cmd/zapi/cmd/index/index.go
@@ -21,14 +21,14 @@ func init() {
 	Index.Add(Create)
 }
 
-type IndexCmd struct {
+type Command struct {
 	*cmd.Command
 }
 
 func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
-	return &IndexCmd{Command: parent.(*cmd.Command)}, nil
+	return &Command{Command: parent.(*cmd.Command)}, nil
 }
 
-func (c *IndexCmd) Run(args []string) error {
+func (c *Command) Run(args []string) error {
 	return charm.ErrNoRun
 }

